doc/man3/OSSL_DISPATCH.pod

```diff
@@ -21,7 +21,11 @@ Arrays of this type are passed between the OpenSSL libraries and the
 providers to describe what functionality one side provides to the other.
 
 Arrays of this type must be terminated with a tuple having function identity
-zero and function pointer NULL.
+zero and function pointer NULL, i.e.,
```

Comment:
Perhaps don't mention the `{ 0, NULL }` option at all.  I.e. _Arrays of this type must be terminated with the L<OSSL_DISPATCH_END(3)> macro._

There doesn't appear to be any documentation for OSSL_DISPATCH_END, so the link isn't useful: _Arrays of this type must be terminated with the OSSL_DISPATCH_END macro._

Note: the macro itself should be documented.  I think it would be sufficient to mention it in the NAME section and have `#define OSSL_DISPATCH_END` in the SYNOPSIS.  The description here is adequate IMO.

Reply:
You mean apart from with the OSSL_DISPATCH_END being explained in the DESCRIPTION section, it should also be explained in the NAME and SYNOPSIS too?

Reply:
Not so much explained, just present.  In NAME, it would just be `OSSL_DISPATCH, OSSL_DISPATCH_END` instead of `OSSL_DISPATCH`.  In SYNOPSIS, a line ` #define OSSL_DISPATCH_END` would be sufficient.

`OSSL_PARAM_END` is documented in OSSL_PARAM_int.pod.  OSSL_PARAM.pod is describing the core parameter kernel -- the structure and nothing more.  OSSL_PARAM_int.pod is describing the helper functions to process parameters and parameter arrays.  We could have put `OSSL_PARAM_END` in either I guess, but it ended up where it ended up.

Reply:
Ok got it. But my doubt is that in the NAME section of file OSSL_DISPATCH.pod, there alread is a line `OSSL_DISPATCH - OpenSSL Core type to define a dispatchable function table` which looks similar to OSSL_PARAM.pod NAME section. It just described the core structure `OSSL_DISPATCH`. So should I just put `OSSL_DISPATCH, OSSL_DISPATCH_END` below that line within the NAME section?

Reply:
Yes.

Reply:
Cool. Thanks.

Reply:
I find the *must* for using the macro unnecessarily harsh.  It could just as well have been presented as a convenience macro, that would have been absolutely fine.  But oh well...